python/mlx/nn/layers/activations.py

```diff
@@ -56,6 +65,15 @@ def relu6(x):
     return mx.minimum(mx.maximum(x, 0), 6.0)
 
 
+def softmax(x, axis: Optional[int] = None):
+    r"""Applies the Softmax function.
+
+    Applies :math:`\frac{e^{x_i}}{\sum_j e^{x_j}}` element wise.
+    """
+    axis = -1 if axis is None else axis
```

Comment:
Why not default `axis=-1` and remove the `Optional`? Seems simpler + you can remove the check for `None` below...

Reply:
Yup makes sense.